scripts/projects.js

```diff
@@ -1,32 +1,61 @@
-var projects = [];
+(function(module){
 
 function Project (opts) {
-  this.title = opts.title;
-  this.category = opts.category;
-  this.developer = opts.developer;
-  this.developerUrl = opts.developerUrl;
-  this.publishedOn = opts.publishedOn;
-  this.body = opts.body;
+  for (keys in opts) {
+    this[keys] = opts[keys];
+  }
 };
 
+Project.all = [];
+
 // Incorporate handlebars.js into code
-Project.prototype.toHtml = function() {
-  var source = $('#project-template').html();
+Project.prototype.toHtml = function(headScriptTemplate) {
+  var source = $(headScriptTemplate).html();
   var template = Handlebars.compile(source);
   var html = template(this);
-
   return html;
 };
 
-myLocalData.sort(function(currentElement, nextElement) {
-  return (new Date(nextElement.publishedOn)) - (new Date(currentElement.publishedOn));
-});
+Project.loadAll = function(dataToPassIn) {
+  dataToPassIn.sort(function(currentElement, nextElement) {
+    return (new Date(nextElement.publishedOn)) - (new Date(currentElement.publishedOn));
+  }).forEach(function(element) {
+    Project.all.push(new Project(element));
+  });
+};
+
+Project.fetchAll = function() {
+  if (localStorage.blogProjects) {
+    // retrieve data from local storage, if available
+    var localStoredData = JSON.parse(localStorage.getItem('blogProjects'));
+    console.log('data in local storage', localStoredData);
+    Project.loadAll(localStoredData);
+    projectView.renderIndexPage();
+  } else {
+    // if data is not in local storage, call Ajax to pull data from blogProjects.json
+    $.ajax({
+      url: '/data/blogProjects.json',
+      method: 'GET',
+      success: successHandler,
+      error: errorHandler
+    });
+  }
+    console.log('calling Ajax the man');
+};
 
-myLocalData.forEach(function(element) {
-  projects.push(new Project(element));
-});
+function successHandler(data) {
+  localStorage.setItem('/data/blogProjects.json', JSON.stringify(data));
```

Comment:
You can name the localstorage key whatever you want, but typically it will look more like a regular variable name than a url path.